### DISCUSSION TITLE
Default support for Linear Advance

### DISCUSSION THREAD

**Discussion by yeawellwhatever** [mriscoc/Ender3V2S1#64] [Q&A] [answered]
Are there any plans to enable Linear Advance support by default in the future?

**Comment by mriscoc** [comment, marked as answer]
EDIT: Linear advance (LA) is now also supported for standalone stepper drivers in the latest firmware versions, read here: https://github.com/mriscoc/Special_Configurations/releases/tag/LinearAdvance

Old answer:
Linear advance (LA) is unsupported in stock Creality boards because the drivers work in `standalone mode`, LA needs that the board drivers work in `UART` mode. So, there will be no default support for LA.  

![linear-advance](https://user-images.githubusercontent.com/2745567/155760607-6b54f875-9488-4b80-9c9e-5194e933cc95.jpg)

**Reply by mhweiner** [reply]
Now that we have support for the SKR E3, can we get Linear Advance? It's supported on that board, and what's what I'm running, and why I actually made the switch from Jyers to Professional. Thanks for all your hard work! Your firmware is amazing!

**Reply by mriscoc** [reply]
I will enable support for linear advanced in the firmware but only for compile from sources.

**Reply by mhweiner** [reply]
Thanks!

**Reply by mhweiner** [reply]
@mriscoc is there anything I need to do besides enabling this in configurations_adv.h? I'm sorry, I'm not entirely sure what you meant when you said you would enable but only for compile from source. Isn't that the same thing?

**Reply by mriscoc** [reply]
I could maybe add some menu items for some parameter, but those items won't be available in regular precompiled binaries.

**Reply by mhweiner** [reply]
Ahh, I see, that makes sense. I haven't played with LA at all yet, but when I do I'll let you know how it goes with the SKR E3 v3.0.

**Reply by ghost** [reply]
@mhweiner any updates? Is your LA build going good along with SKR mini e3 v3? Have you stay or moved to Klipper?
Have you achieved Sensorless homing?

**Reply by mriscoc** [reply]
Linear Advance versions are available here: https://github.com/mriscoc/Special_Configurations/releases/tag/LinearAdvance

**Reply by mhweiner** [reply]
@badaszszsz I am currently running a custom build of Marlin 2.1.2 with LA. I've also completely removed the Ender 3 V2 display in favor of a basic FYSETC Mini 12864 Neopixel display. Works great. Eventually, I'll move to Klipper.

**Reply by phantum29** [reply]
I've modified my 4.2.7 board to have all 4 drivers connected for UART, and detached the diag pins. Do I need to modify the firmware and compile it for the pins I've used? Then, to take advantage of having UART for the stepper drivers, do I need to add/configure anything for features like LA and vref, enable/disable spread cycle etc. I suspect because you have a skr mini e3 config, that it might be baked in, but disabled due to creality's board design. I'm comparing your skr mini config to the 4.2.7, but thought I'd ask in case I miss something. TIA!

**Reply by mriscoc** [reply]
You need to change the drivers settings in Configuration.h file and build a custom version.
